File: scala/Factorial.scala
object Factorial {

	def fact(num: Int) : BigInt  = {
	  		if(num==1){
	  			Thread.dumpStack()
	  			1
	  		} else {
	  			 fact(num-1)
	  		}
	  	}

	def main(args: Array[String]): Unit = {
		println(fact(20000))
	}
	
}

File: java/Factorial.java
public class Factorial {

	public long fact(int num){
		if(num == 1) {
			Thread.dumpStack();
			return 1;
		}

		else
			return fact(num-1);
	}

	public static void main(String args[]){
		System.out.println(new Factorial().fact(20000));
	}

}